src/lib/parallel_scan/scan_state.md

```diff
@@ -0,0 +1,141 @@
+# Scan State Refactoring
+
+The parallel scan state currently is a full-binary tree with leaves (or called `Base` nodes) having values of type `Transaction_with_witness.t` and intermediate nodes (or called `Merge` nodes) having values of type `Ledger_proof_with_sok_message.t`
```

Comment:
does "currently" here mean the data structure is changing? That's what is suggested, but I don't think that is meant.